### PR TITLE
Update porting guide to mention no_log change

### PR DESCRIPTION
**Pull request by willthames**
##### SUMMARY
Document that `no_log` now needs to be templated

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
no_log

### CODE REVIEW

**Comment by samdoran** [comment]
This does not seem to be specific to `no_log`. The problem with bare boolean variables in already documented in the [2.8 porting guide](https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_2.8.html#bare-variables-in-conditionals). I'm not sure this needs be documented in the 2.10 porting guide. Seems like the incorrect place.

**Comment by willthames** [comment]
@samdoran - that makes sense, but that change is a big obvious note in the 2.8 guide, but I don't expect to read that to understand a breaking change in 2.10. This PR isn't the right fix (perhaps copying the 2.8 note into 2.10 might be).

**Comment by willthames** [comment]
I can't see the mention of bare boolean variables in the 2.10 guide at all (`grep bare` returns no results)

**Comment by willthames** [comment]
@samdoran ready for rereview

File: docs/docsite/rst/porting_guides/porting_guide_2.10.rst [hunk 1]
```diff
@@ -28,7 +28,7 @@ Playbook
 * A new fact, ``ansible_processor_nproc`` reflects the number of vcpus
   available to processes (falls back to the number of vcpus available to
   the scheduler).
-
+* ``conditional_bare_variables`` now defaults to ``false`` (so that ``when: result`` will always default to ``true`` if ``result`` is a non-empty string, even if that string is ``false``). See the :ref:`2.8 porting guide note <bare_variables_in_conditionals>` for more details.
```
Comment (bcoca) [inline docs/docsite/rst/porting_guides/porting_guide_2.10.rst:31]: this is not the issue, in any case, if its already in previous porting guide, we don't need to copy in every subsequent one.

The fix that causes this is not related to 'bare' variables but to correctly typing 'boolean' keywords, which previously accepted incorrect values that ended up 'truthy' in weird ways.

**Review by bcoca — CHANGES_REQUESTED** [review]

File: docs/docsite/rst/porting_guides/porting_guide_2.10.rst (hunk 1 above)
Comment (willthames) [reply docs/docsite/rst/porting_guides/porting_guide_2.10.rst:31]: This is all semantics. If you folks want to word it better as to why my playbooks break with 2.10 but work in 2.9, be my guest.

Comment (bcoca) [reply docs/docsite/rst/porting_guides/porting_guide_2.10.rst:31]: if you can confirm that #67625 is the cause, i'll write up the porting guide entry

Comment (willthames) [reply docs/docsite/rst/porting_guides/porting_guide_2.10.rst:31]: @bcoca: confirmed

```
- hosts: localhost
  connection: local
  vars:
    unsafe_show_logs: false

  tasks:
    - debug:
        msg: hello
      no_log: not unsafe_show_logs
```

That playbook works with `git checkout babac66f9cd978b765f58a21c20f3577308f5504^` (i.e. the commit before that one) but not with `git checkout babac66f9cd978b765f58a21c20f3577308f5504`

Comment (bcoca) [reply docs/docsite/rst/porting_guides/porting_guide_2.10.rst:31]: @willthames  thanks, sorry for the inconvenience, I had not foreseen that many playbooks using the 'broken behaviour', which is why i had not added an extensive porting guide entry (there is one about the change, but maybe not doing good job about explaining the implications). 

Correcting that now.

FYI, no_log was not being 'auto templated' it was just equating any strings to 'True'

**Comment by samdoran** [comment]
So it seems this was not related to bare variables in conditionals but due to the change in https://github.com/ansible/ansible/commit/babac66f9cd978b765f58a21c20f3577308f5504. Do the changes in #69948 address this issue by explaining what's going on? If so, can we close this in favor of that PR?

**Comment by samdoran** [comment]
Sorry for not fully understanding the full problem.

**Comment by bcoca** [comment]
closing in favor of #69948

**Pull request closed without merging**